test/unit/ngsiv2/amqpBinding-test.js

```diff
@@ -0,0 +1,320 @@
+                done();
+            }, 100);
+        });
+    });
+
+    describe('When a new measure arrives for an unprovisioned Device', function() {
+        var groupCreation = {
+            url: 'http://localhost:4041/iot/services',
+            method: 'POST',
+            json: utils.readExampleFile('./test/unit/ngsiv2/groupProvisioning/provisionFullGroupAMQP.json'),
+            headers: {
+                'fiware-service': 'TestService',
+                'fiware-servicepath': '/testingPath'
+            }
+        };
+
+        beforeEach(function(done) {
+            // Note: /v1/updateContext response is not processed by IOTA so its content is irrelevant,
+            // as far as it is a 200 OK
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v1/updateContext')
+                .reply(200, '{}');
+
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'TestService')
+                .matchHeader('fiware-servicepath', '/testingPath')
+                .post('/v2/entities/SensorMachine:UL_UNPROVISIONED/attrs',
+                 utils.readExampleFile('./test/unit/ngsiv2/contextRequests/unprovisionedMeasure.json'))
+                .reply(204);
+
+            request(groupCreation, function(error, response, body) {
+                done();
+            });
+        });
+
+        it('should send a new update context request to the Context Broker with just that attribute', function(done) {
+            channel.publish(config.amqp.exchange, '.80K09H324HV8732.UL_UNPROVISIONED.attrs.temperature',
+                new Buffer('23'));
+
+            setTimeout(function() {
+                contextBrokerMock.done();
+                done();
+            }, 100);
+        });
+    });
+
+    describe('When a new multiple measure arrives to a Device routing key with one measure', function() {
+        beforeEach(function() {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/singleMeasure.json'))
+                .reply(204);
+        });
+
+        it('should send a single update context request with all the attributes', function(done) {
+            channel.publish(config.amqp.exchange, '.1234.MQTT_2.attrs', new Buffer('temperature|23'));
+
+            setTimeout(function() {
+                contextBrokerMock.done();
+                done();
+            }, 100);
+        });
+    });
+
+    describe('When a new multiple measure arrives to a Device routing key with a faulty payload', function() {
+        beforeEach(function() {
+            contextBrokerMock
```

Comment:
In the case of faulty measure I understand that no `POST /v2/entities/Second%20UL%20Device/attrs` is sent, so contextBrokerMock() doesn't need this configuration, it could be removed, isn't it?

Reply:
The commit mentioned in "Fixed in" (93cfad4) doesn't touch amqpBinding-test.js file.

Please, try to be more precise with the "Fixed in" messages... with a large PR like this one, it is a bit confusing to re-review if references are wrong.

Reply:
NTC... sorry for the mistake

Reply:
Let's check if I understand this test case correctly...

The IOTA  has been provisioned with test/unit/ngsiv2/deviceProvisioning/provisionDeviceAMQP1.json:

```
{
  "devices": [
    {
      "protocol": "GENERIC_PROTO",
      "transport": "AMQP",
      "device_id": "MQTT_2",
      "entity_name": "Second UL Device",
      "entity_type": "AnMQTTDevice",
      "timezone": "America/Santiago",
      "attributes": [
        {
          "name": "temperature",
          "type": "celsius"
        },
        {
          "name": "humidity",
          "type": "degrees"
        }
      ]
    }
  ]
}
```
Then, a "faulty" measure arrives, i.e. the "notAULPayload " string is not correct UL syntax. As a consecuence, the IOTA doens't progress any update toward CB (as much, it will print a WARN or ERROR trace in the logs, I guess).

Is my interpretation correct?

Reply:
yes it is :)

Reply:
So... declaring that CB will receive a `POST /v2/entities/Second%20UL%20Device/attrs` request wouldn't match the test case, right?

That's the point of removing the mock stuf...

Reply:
So do you mean to remove all of this test case ? 

```
describe('When a new multiple measure arrives to a Device routing key with a faulty payload', function() {
```
As you said before, The IOTA print A WARN:

```
 msg=MEASURES-003: Parse error parsing incoming MQTT message [%] | comp=IoTAgent notAULPayload
```

Reply:
Not al of the test case, only the mock expectations (i.e. remove L180-L186). If after removing that block, test are still passing, that would show that the mock wasn't needed.

In additon, it would be great to check such log trace is generated as part of the test case (as my other comment points out, i.e. https://github.com/telefonicaid/iotagent-ul/pull/242/files#r170203708).

Reply:
...or, if you don't find any way off doint it easily, at least a FIXME/TODO mark in that point (just before L192, withing `setTimeout()`) defering the check to a future improvement in the test.

Reply:
Test was passed in spite of remove lines L180 - L186.  Added FIXEME/TODO mark in section L192. 

See here: 202481507cb55718ea206dbdec93a421e80e9bf2